src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx

```diff
@@ -116,15 +84,41 @@ export function DocTableLegacy(renderProps: DocTableLegacyProps) {
                 render-complete
                 use-new-fields-api="useNewFieldsApi"
                 sorting="sort"></doc_table>`,
-    },
-    () => getServices().getEmbeddableInjector()
-  );
+  };
+
+  return async () => {
+    try {
+      const injector = await getServices().getEmbeddableInjector();
+      return await injectAngularElement(domNode, directive.template, props, injector);
+    } catch (e) {
+      render(<div>error</div>, domNode);
+      return () => void 0;
+    }
+  };
+}
+
+export function DocTableLegacy(renderProps: DocTableLegacyProps) {
   const ref = useRef<HTMLDivElement>(null);
+  const scope = useRef<AngularScope>();
+
   useEffect(() => {
-    if (ref && ref.current) {
-      return renderFn(ref.current, renderProps);
+    if (ref && ref.current && !scope.current) {
+      const fn = getRenderFn(ref.current, renderProps);
+      fn().then((newScope) => {
+        scope.current = newScope;
+      });
+    } else if (scope && scope.current) {
+      scope.current = Object.assign(scope.current, renderProps);
```

Comment:
I wonder if we should store the `renderProps` as `$scope.renderProps` rather than merging them into scope. We'd need to update the template to access `renderProps.searchDescription`, etc., but then we wouldn't have to worry about props that weren't passed on subsequent renders, they would automatically be deleted.

Reply:
thx, I've improved the code, now hopefully the functional failing test also should work